lib/streamlit/elements/widgets/button_group.py

```diff
@@ -78,28 +86,38 @@
 # in base64 format and send it over the wire as an image.
 _SELECTED_STAR_ICON: Final = ":material/star_filled:"
 
+SelectionMode: TypeAlias = Literal["single", "multi"]
+
 
-class FeedbackSerde:
+class SingleSelectSerde(Generic[T]):
     """Uses the MultiSelectSerde under-the-hood, but accepts a single index value
-    and deserializes to a single index value. This is because for feedback, we always
-    allow just a single selection.
+    and deserializes to a single index value.
+    This is because button_group can be single and multi select, but we use the same
+    proto for both and, thus, map single values to a list of values and a receiving
+    value wrapped in a list to a single value.
 
-    When a sentiment_mapping is provided, the sentiment corresponding to the index is
-    serialized/deserialized. Otherwise, the index is used as the sentiment.
+    When a default_value is provided is provided, the option corresponding to the
+    index is serialized/deserialized.
     """
 
-    def __init__(self, option_indices: list[int]) -> None:
+    def __init__(
+        self,
+        option_indices: Sequence[T],
+        default_value: list[int] | None = None,
+    ) -> None:
         """Initialize the FeedbackSerde with a list of sentimets."""
-        self.multiselect_serde: MultiSelectSerde[int] = MultiSelectSerde(option_indices)
+        self.multiselect_serde: MultiSelectSerde[T] = MultiSelectSerde(
```

Comment:
Is this expected that the `SingleSelectSerde` is using `MultiSelectSerde` internally. If yes, maybe add a comment explaining why.

Reply:
Yep! We use it so that we can use the same proto in the background. The docstring has an explanation for it. I added a comment here to look at the docstring